src/data_provider/src/osinfo/sysOsParsers.cpp

```diff
@@ -290,14 +278,14 @@ bool GentooOsParser::parseFile(std::istream& in, nlohmann::json& output)
 bool SuSEOsParser::parseFile(std::istream& in, nlohmann::json& output)
 {
     constexpr auto SEPARATOR{'='};
-    static const std::map<std::string, std::string> KEY_MAP
+    static const std::vector<std::pair<std::string, std::string>> KEY_MAPPING
```

Comment:
The changes made to this function, while elementary, exceed the original scope of the issue